### PR TITLE
Rename integration test ios_app_with_watch_companion -> ios_app_with_extensions

### PR DESCRIPTION
**Pull request by jmagman**
## Description

Rename ios_app_with_watch_companion to something more generic so it can be repurposed for more iOS project extensions.
- Rename folder
- Rename test file and manifest task name
- Update README
- Update bundle identifiers from com.example.watchCompanionTest to io.flutter.extensionTest

## Related Issues

Will use this test to add a Swift Package integration test in a future PR (https://github.com/flutter/flutter/issues/57083)

## Checklist
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change
- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*

### CODE REVIEW

**Comment by christopherfujino** [comment]
I re-ran the windows commands test, as it timed out

File: dev/devicelab/manifest.yaml [hunk 1]
```diff
@@ -685,9 +685,9 @@ tasks:
 #    required_agent_capabilities: ["mac/ios"]
 #    flaky: true
 
-  ios_app_with_watch_companion:
+  ios_app_with_extensions_test:
     description: >
-      Checks that an iOS app with a watchOS companion can be build for physical and simulated devices.
+      Checks that an iOS app with extensions can be build for physical and simulated devices.
```
Comment (christopherfujino) [inline dev/devicelab/manifest.yaml:690]:
```suggestion
      Checks that an iOS app with extensions can be built for physical and simulated devices.
```

File: dev/integration_tests/ios_app_with_extensions/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# iOS Extensions Test
+
+Integration test for to test building iOS app with
```
Comment (christopherfujino) [inline dev/integration_tests/ios_app_with_extensions/README.md:3]:
```suggestion
Integration test to test building an iOS app with
```

File: dev/integration_tests/ios_app_with_extensions/ios/Runner.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -630,7 +629,7 @@
 				MTL_ENABLE_DEBUG_INFO = INCLUDE_SOURCE;
 				MTL_FAST_MATH = YES;
 				OTHER_LDFLAGS = "";
-				PRODUCT_BUNDLE_IDENTIFIER = com.example.watchCompanionTest.watchkitapp;
+				PRODUCT_BUNDLE_IDENTIFIER = io.flutter.extensionTest.watchkitapp;
```
Comment (christopherfujino) [inline dev/integration_tests/ios_app_with_extensions/ios/Runner.xcodeproj/project.pbxproj:632]: this should probably be `dev.flutter.extensionTest.watchkitapp`, but that's probably a bigger refactor for another day

**Review by christopherfujino — APPROVED** [review]
LGTM with two spelling nits.

**Comment by jmagman** [comment]
All tests passed before my spelling tweaks, merging.

**Pull request merged**